cloud: serverless HA (#19500)
pingcap/docs: tidb-cloud/serverless-high-availability.md

Reviewer: Preview: https://pingcap-docsite-preview-git-preview-cl-d5eb3d-oreoxmts-projects.vercel.app/tidbcloud/serverless-high-availability

TiDB Cloud Serverless ensures business continuity with minimal service disruption. In the event of primary zone failure:

- New replicas of Gateway and TiDB are automatically created in the standby availability zone.
- The elastic load balancer detects the active gateway replicas in the standby availability zone and redirects the OLTP traffic from the failed primary zone.

In addition to high availability through TiKV replication, TiKV instances are deployed and configured to place each data replica in a different availability zone. As long as two zones are operating normally, the system remains available to serve your application. Data persistence is ensured by regularly pushing data to S3 for high durability. Even if two zones fail, your data on S3 remains accessible and can be recovered.

Applications remain unaffected by non-primary zone failures and are unaware of such issues. In the event of a primary zone failure, both Gateway and TiDB are launched in the standby zone to handle the workload. Applications should implement retry logic for connections to redirect new requests to the active servers in the standby zone.

> ## Automatic Backups and Durability

Reviewer: ```suggestion

## Automatic Backups and Durability
```

---
title: High Availability in TiDB Cloud Serverless
summary: Learn about the high availability architecture of TiDB Cloud Serverless. Discover Zonal and Regional High Availability options, automated backups, failover processes, and how TiDB ensures data durability and business continuity.
---

# Overview of High Availability in TiDB Cloud Serverless

> TiDB Cloud Serverless is designed with robust mechanisms to maintain high availability and data durability by default, preventing single points of failure and ensuring continuous service even in the face of disruptions. As a fully managed service based on the battle-tested TiDB Open Source product, it inherits TiDB's core high availability (HA) features and augments them with additional cloud-native capabilities.

Reviewer: ```suggestion
TiDB Cloud Serverless provides built-in high availability and data durability through its distributed architecture. Based on the open-source TiDB database, it combines TiDB's native high availability features with cloud-native capabilities to prevent single points of failure and maintain service continuity during disruptions.
```

> TiDB provides high availability and data durability using the Raft consensus algorithm, ensuring consistent replication of all data changes across multiple nodes. This allows TiDB to continue servicing both read and write requests, even during node failures or network partitions, guaranteeing high data durability and fault tolerance. TiDB Cloud Serverless enhances these foundational capabilities by offering two types of high availability to meet different operational requirements:

Reviewer: ```suggestion
TiDB ensures high availability and data durability using the Raft consensus algorithm. This algorithm consistently replicates data changes across multiple nodes, allowing TiDB to handle read and write requests even in the event of node failures or network partitions. This approach provides both high data durability and fault tolerance.

TiDB Cloud Serverless extends these capabilities with two types of high availability to meet different operational requirements:
```

> - **Zonal (Default)**: This option places all nodes in the same availability zone, resulting in lower network latency. It ensures high availability without requiring application redundancy across zones, making it ideal for applications achieving the highest availability within a single zone with minimal latency. Available in all regions supporting TiDB Cloud Serverless.

Reviewer: ```suggestion
- **Zonal high availability (default)**: This option places all nodes within a single availability zone, reducing network latency. It ensures high availability without requiring application-level redundancy across zones, making it suitable for applications that prioritize low latency within a single zone. Zonal high availability is available in all regions that support TiDB Cloud Serverless. For more information, see [Zonal high availability architecture](#zonal-high-availability-architecture).
```

> - **Regional (Beta)**. This option distributes nodes across multiple availability zones, offering maximum infrastructure isolation and redundancy. While it provides the highest level of availability, it requires configuring application-level redundancy across zones. Recommended for organizations that requires the highest availability against infrastructure failures within a zone at the cost of increased latency and potential cross-zone data transfer fees. Available  in selected regions supporting multiple availability zones and can only be enabled during cluster creation.

Reviewer: ```suggestion
- **Regional high availability (beta)**: This option distributes nodes across multiple availability zones, offering maximum infrastructure isolation and redundancy. It provides the highest level of availability but requires application-level redundancy across zones. It is recommended to choose this option if you need maximum availability protection against infrastructure failures within a zone. Note that it increases latency and might incur cross-zone data transfer fees. This feature is available in selected regions with multi-availability zone support and can only be enabled during cluster creation. For more information, see [Regional high availability architecture](#regional-high-availability-architecture).
```

> **Note:**
>
> Regional HA is in Beta and only avliable in AWS Tokyo (ap-northeast-1) region.

> ## Zonal High Availability Architecture

Reviewer: ```suggestion
## Zonal high availability architecture
```

> When you create a cluster with default Zonal High Availability, all components, including Gateway, TiDB, TiKV, and TiFlash compute/write nodes, reside within the same avaliability zone. The placement of these components in the data plane offer infrastructure redundancy with virtual machine pools, which minimizes failover time and network latency due to colocation.
> 
> > **Note:**
> >
> > This is the default option and is available in all AWS regions that support TiDB Cloud Serverless.

Reviewer: ```suggestion
> **Note:**
>
> Zonal high availability is the default option and is available in all AWS regions that support TiDB Cloud Serverless.

When you create a cluster with the default zonal high availability, all components, including Gateway, TiDB, TiKV, and TiFlash compute/write nodes, run in the same availability zone. The placement of these components in the data plane offer infrastructure redundancy with virtual machine pools, which minimizes failover time and network latency due to colocation.
```

![Serverless Regional HA](../media/tidb-cloud/serverless-regional-high-avaliability-aws.png)

> In Zonal High Availability architecture:

Reviewer: ```suggestion
In zonal high availability architecture:
```

> - The Placement Driver (PD) is deployed across multiple availability zones, and data is always replicated redundantly across zones to ensure the highest level of availability.
> - Data is replicated across TiKV servers and TiFlash write nodes within the local availability zone.
> - TiDB servers and TiFlash compute nodes read from and write to these TiKV and TiFlash write nodes, which are safeguarded by storage-level replication.

Reviewer: ```suggestion
- The Placement Driver (PD) is deployed across multiple availability zones, ensuring high availability by replicating data redundantly across zones.
- Data is replicated across TiKV servers and TiFlash write nodes within the local availability zone.
- TiDB servers and TiFlash compute nodes read from and write to TiKV and TiFlash write nodes, which are safeguarded by storage-level replication.
```

> ### Failover Process

Reviewer: ```suggestion
### Failover process
```

> TiDB Cloud Serverless ensures that failover remains transparent to your applications. In the event of a failover:

Reviewer: ```suggestion
TiDB Cloud Serverless ensures a transparent failover process for your applications. During a failover:
```

- A new replica is created to replace the failed one.

> - Servers offering storage services recover local cache from the persisted data on S3 and bring it online to a consistent state with their replicas.

Reviewer: ```suggestion
- Servers providing storage services recover local caches from persisted data on Amazon S3, restoring the system to a consistent state with the replicas.
```

> In the storage layer, persisted data is regularly pushed to S3 for high durability. Moreover, immediate updates are not only replicated across multiple TiKV servers but also stored on the EBS of each server, which further replicates the data for even greater durability. TiDB automatically handles any issues by backing off and retrying in milliseconds. This ensures that the failover process remains transparent to the client application, requiring no intervention from the user.

Reviewer: ```suggestion
In the storage layer, persisted data is regularly pushed to Amazon S3 for high durability. Moreover, immediate updates are not only replicated across multiple TiKV servers but also stored on the EBS of each server, which further replicates the data for additional durability. TiDB automatically resolves issues by backing off and retrying in milliseconds, ensuring the failover process remains seamless for client applications.
```

> The gateway and computing layer are stateless, so failover is as simple as restarting them immediately elsewhere. Applications should use retry logic for their connections. Though the Zonal setup provides excellent availability, it is limited if the entire zone is down, which would result in downtime until dependent services in that zone are restored.

Reviewer: ```suggestion
The gateway and computing layers are stateless, so failover involves restarting them elsewhere immediately. Applications should implement retry logic for their connections. While the zonal setup provides high availability, it cannot handle an entire zone failure. If the zone becomes unavailable, downtime will occur until the zone and its dependent services are restored.
```

> ## Regional High Availability Architecture

Reviewer: ```suggestion
## Regional high availability architecture
```

> When you create a cluster with Regional High Availability, critical OLTP workload components such as Placement Driver (PD) and TiKV are deployed across multiple availability zones to ensure redundant replication and maximizing availability. During normal operations, components like Gateway, TiDB, and TiFlash compute/write nodes are located within the Primary Availability Zone. These components in data plane offer infrastructure redundancy through virtual machine pools, which minimizes failover time and network latency due to colocation.

Reviewer: ```suggestion
When you create a cluster with regional high availability, critical OLTP (Online Transactional Processing) workload components, such as PD and TiKV, are deployed across multiple availability zones to ensure redundant replication and maximizing availability. During normal operations, components like Gateway, TiDB, and TiFlash compute/write nodes are hosted in the primary availability zone. These components in data plane offer infrastructure redundancy through virtual machine pools, which minimizes failover time and network latency due to colocation.
```

> > **Note:**
> >
> > Regional HA can only be enabled during cluster creation and is available in selected regions that support multiple availability zones.

Reviewer: ```suggestion
> **Note:**
>
> - Regional high availability is currently in beta and only available in the AWS Tokyo (`ap-northeast-1`) region.
> - You can enable regional high availability only during cluster creation.
```

![Serverless Regional HA](../media/tidb-cloud/serverless-regional-high-avaliability-aws.png)

> In Regional High Availability architecture:

Reviewer: ```suggestion
In regional high availability architecture:
```

- The Placement Driver (PD) and TiKV are deployed across multiple availability zones, and data is always replicated redundantly across zones to ensure the highest level of availability.
- Data is replicated across TiFlash write nodes within the primary availability zone.
- TiDB servers and TiFlash compute nodes read from and write to these TiKV and TiFlash write nodes, which are safeguarded by storage-level replication.

> ### Failover Process

Reviewer: ```suggestion
### Failover process
```

> In the unlikely event of a primary zone failure scenario, which could be caused by a natural disaster, a configuration change, a software issue, or a hardware component failure. Critical OLTP workload components, including Gateway and TiDB, are automatically launched in the Standby Availability Zone. Traffic is then automatically redirected there. This process ensures a swift recovery to normal operations and maintains business continuity.

Reviewer: ```suggestion
In the rare event of a primary zone failure scenario, which could be caused by a natural disaster, configuration change, software issue, or hardware failure, critical OLTP workload components, including Gateway and TiDB, are automatically launched in the standby availability zone. Traffic is automatically redirected to the standby zone to ensure swift recovery and maintain business continuity.
```

> TiDB Cloud Serverless ensures business continuity with minimal service disruption. In the event of primary zone failure:

Reviewer: ```suggestion
TiDB Cloud Serverless minimizes service disruption and ensures business continuity during a primary zone failure by performing the following actions:
```

> - New replicas of Gateway and TiDB are automatically created in the standby availability zone.
> - The elastic load balancer detects the active gateway replicas in the standby availability zone and redirects the OLTP traffic from the failed primary zone.

Reviewer: ```suggestion
- Automatically create new replicas of Gateway and TiDB in the standby availability zone.
- Use the elastic load balancer to detect active gateway replicas in the standby availability zone and redirect OLTP traffic from the failed primary zone.
```

> In addition to high availability through TiKV replication, TiKV instances are deployed and configured to place each data replica in a different availability zone. As long as two zones are operating normally, the system remains available to serve your application. Data persistence is ensured by regularly pushing data to S3 for high durability. Even if two zones fail, your data on S3 remains accessible and can be recovered.

Reviewer: ```suggestion
In addition to providing high availability through TiKV replication, TiKV instances are deployed and configured to place each data replica in a different availability zone. The system remains available as long as two availability zones are operating normally. For high durability, data persistence is ensured by regularly backing up data to S3. Even if two availability zones fail, data stored in S3 remains accessible and recoverable.
```

> Applications remain unaffected by non-primary zone failures and are unaware of such issues. In the event of a primary zone failure, both Gateway and TiDB are launched in the standby zone to handle the workload. Applications should implement retry logic for connections to redirect new requests to the active servers in the standby zone.

Reviewer: ```suggestion
Applications are unaffected by failures in non-primary zones and remain unaware of such events. During a primary zone failure, Gateway and TiDB are launched in the standby availability zone to handle workloads. Ensure that your applications implement retry logic to redirect new requests to active servers in the standby availability zone.
```

> ## Automatic Backups and Durability

Reviewer: ```suggestion
## Automatic backups and durability
```

> Database backups are a critical component of any business continuity and disaster recovery strategy as they help protect your data from corruption or accidental deletion. With these backups, you can restore your database to a specific point in time within the configured retention period, minimizing potential data loss and downtime.

Reviewer: ```suggestion
Database backups are essential for business continuity and disaster recovery, helping to protect your data from corruption or accidental deletion. With backups, you can restore your database to a specific point in time within the retention period, minimizing data loss and downtime.
```

> TiDB Cloud Serverless offers robust, automated backup mechanisms to ensure continuous protection of your data:
> 
> - **Daily Full Backups**: A full backup of your database is taken once a day to capture the entire state of the database.
> - **Continuous Transaction Log Backups**: Transaction logs are backed up continuously, with backups occurring approximately every 5 minutes. The exact frequency may vary based on the level of database activity.

Reviewer: ```suggestion
TiDB Cloud Serverless provides robust automated backup mechanisms to ensure continuous data protection:

- **Daily full backups**: A full backup of your database is created once a day, capturing the entire database state.
- **Continuous transaction log backups**: Transaction logs are backed up continuously, typically every 5 minutes, though the exact frequency depends on database activity.
```

> These automated backups allow you to restore your database either from a full backup or from a specific point-in-time by combining full backups with continuous transaction logs. This flexibility ensures that you can recover your database to a precise point just before an incident occurred. 

Reviewer: ```suggestion
These automated backups enable you to restore your database either from a full backup or from a specific point in time by combining full backups with continuous transaction logs. This flexibility ensures you can recover your database to a precise point just before an incident occurred.
```

> **Note:**
>

> > Automatic backups, including snapshot-based and continuous backups for Point-in-Time Recovery (PiTR), are performed on S3, which offers regional-level high durability.

Reviewer: ```suggestion
> Automatic backups, including snapshot-based and continuous backups for Point-in-Time Recovery (PITR), are performed on Amazon S3, which provides regional-level high durability.
```

> ## Impact on Sessions During Failures

Reviewer: ```suggestion
## Impact on sessions during failures
```

> In the event of a failure, ongoing transactions on the failed server are affected. Although failover remains transparent to your applications by default, applications need to implement logic to handle recoverable failures in the middle of a transaction. Here's what happens in various failure scenarios:

Reviewer: ```suggestion
During a failure, ongoing transactions on the failed server might be interrupted. Although failover is transparent to applications, you must implement logic to handle recoverable failures during active transactions. Different failure scenarios are handled as follows:
```

> - **TiDB Failures**: If a TiDB instance fails, the client connection remains unaffected because TiDB Cloud Serverless automatically re-routes traffic through the gateway. Transactions running on the failed TiDB instance may be interrupted, but the system ensures that committed data is not lost, and new transactions can be handled by another available TiDB instance.
> - **Gateway Failures**: If the Gateway fails, it disrupts the client connection. TiDB Cloud Serverless gateways are stateless, allowing them to be restarted immediately in a new zone or server. The system also automatically redirects traffic to the new gateway, minimizing downtime.

Reviewer: ```suggestion
- **TiDB failures**: If a TiDB instance fails, client connections are unaffected because TiDB Cloud Serverless automatically reroutes traffic through the gateway. While transactions on the failed TiDB instance might be interrupted, the system ensures that committed data is preserved, and new transactions are handled by another available TiDB instance.
- **Gateway failures**: If the Gateway fails, client connections are disrupted. However, TiDB Cloud Serverless gateways are stateless and can restart immediately in a new zone or server. Traffic is automatically redirected to the new gateway, minimizing downtime.
```

> Applications should handle recoverable failures with retry logic. For specific implementation details and usage, refer to the documentation for your respective driver or ORM (e.g., [JDBC](https://dev.mysql.com/doc/connector-j/en/connector-j-config-failover.html)).

Reviewer: ```suggestion
To handle recoverable failures, add retry logic to your application. For implementation details, refer to your driver or ORM documentation (for example, [JDBC](https://dev.mysql.com/doc/connector-j/en/connector-j-config-failover.html)).
```

## RTO and RPO

> As you develop your business continuity plan, understand the maximum acceptable time before the application fully recovers after the disruptive event. The time required for an application to fully recover is known as the Recovery Time Objective (RTO). Also understand the maximum period of recent data updates (time interval) the application can tolerate losing when recovering from an unplanned disruptive event. The potential data loss is known as Recovery Point Objective (RPO).

Reviewer: ```suggestion
When creating your business continuity plan, consider these two key metrics:

- Recovery Time Objective (RTO): The maximum acceptable time it takes for the application to fully recover after a disruptive event.
- Recovery Point Objective (RPO): The maximum acceptable time interval of recent data updates that the application can tolerate losing during recovery from an unplanned disruptive event.
```

> The following table compares RPO and RTO of each high availability option:

Reviewer: ```suggestion
The following table compares the RTO and RPO for each high availability option:
```

> | High Availability Architecture | RTO (downtime)                | RPO (data loss) |
> |--------------------------------|-------------------------------|-----------------|
> | Zonal                          | Near 0                        | 0               |
> | Regional                       | Typically less than 600 seconds | 0               |

Reviewer: ```suggestion
| High availability architecture | RTO (downtime)                | RPO (data loss) |
|--------------------------------|-------------------------------|-----------------|
| Zonal high availability                         | Near 0 seconds                      | 0               |
| Regional high availability                      | Typically less than 600 seconds | 0               |
```

---
title: High Availability in TiDB Cloud Serverless
summary: Learn about the high availability architecture of TiDB Cloud Serverless. Discover Zonal and Regional High Availability options, automated backups, failover processes, and how TiDB ensures data durability and business continuity.
---

> # Overview of High Availability in TiDB Cloud Serverless

Reviewer: ```suggestion
# High Availability in TiDB Cloud Serverless
```
Update the H1 heading to match the `title` specified in the metadata section.

TiDB Cloud Serverless is designed with robust mechanisms to maintain high availability and data durability by default, preventing single points of failure and ensuring continuous service even in the face of disruptions. As a fully managed service based on the battle-tested TiDB Open Source product, it inherits TiDB's core high availability (HA) features and augments them with additional cloud-native capabilities.

> 

Reviewer: ```suggestion

## Overview

```

- **Zonal (Default)**: This option places all nodes in the same availability zone, resulting in lower network latency. It ensures high availability without requiring application redundancy across zones, making it ideal for applications achieving the highest availability within a single zone with minimal latency. Available in all regions supporting TiDB Cloud Serverless.

- **Regional (Beta)**. This option distributes nodes across multiple availability zones, offering maximum infrastructure isolation and redundancy. While it provides the highest level of availability, it requires configuring application-level redundancy across zones. Recommended for organizations that requires the highest availability against infrastructure failures within a zone at the cost of increased latency and potential cross-zone data transfer fees. Available  in selected regions supporting multiple availability zones and can only be enabled during cluster creation.

> **Note:**
>

> > Regional HA is in Beta and only avliable in AWS Tokyo (ap-northeast-1) region.

Reviewer: ```suggestion
> Regional high availability is in beta and only available in the AWS Tokyo (`ap-northeast-1`) region.
```

## Zonal High Availability Architecture

When you create a cluster with default Zonal High Availability, all components, including Gateway, TiDB, TiKV, and TiFlash compute/write nodes, reside within the same avaliability zone. The placement of these components in the data plane offer infrastructure redundancy with virtual machine pools, which minimizes failover time and network latency due to colocation.

> **Note:**
>
> This is the default option and is available in all AWS regions that support TiDB Cloud Serverless.

> ![Serverless Regional HA](../media/tidb-cloud/serverless-regional-high-avaliability-aws.png)

Reviewer: ```suggestion
![TiDB Cloud Serverless regional high availability](/media/tidb-cloud/serverless-regional-high-avaliability-aws.png)
```

The gateway and computing layer are stateless, so failover is as simple as restarting them immediately elsewhere. Applications should use retry logic for their connections. Though the Zonal setup provides excellent availability, it is limited if the entire zone is down, which would result in downtime until dependent services in that zone are restored.

## Regional High Availability Architecture

When you create a cluster with Regional High Availability, critical OLTP workload components such as Placement Driver (PD) and TiKV are deployed across multiple availability zones to ensure redundant replication and maximizing availability. During normal operations, components like Gateway, TiDB, and TiFlash compute/write nodes are located within the Primary Availability Zone. These components in data plane offer infrastructure redundancy through virtual machine pools, which minimizes failover time and network latency due to colocation.

> **Note:**
>
> Regional HA can only be enabled during cluster creation and is available in selected regions that support multiple availability zones.

> ![Serverless Regional HA](../media/tidb-cloud/serverless-regional-high-avaliability-aws.png)

Reviewer: ```suggestion
![TiDB Cloud Serverless regional high availability](/media/tidb-cloud/serverless-regional-high-avaliability-aws.png)
```

In Regional High Availability architecture:

- The Placement Driver (PD) and TiKV are deployed across multiple availability zones, and data is always replicated redundantly across zones to ensure the highest level of availability.
- Data is replicated across TiFlash write nodes within the primary availability zone.
- TiDB servers and TiFlash compute nodes read from and write to these TiKV and TiFlash write nodes, which are safeguarded by storage-level replication.

> 

Reviewer: ```suggestion
```

TiDB Cloud Serverless ensures business continuity with minimal service disruption. In the event of primary zone failure:

- New replicas of Gateway and TiDB are automatically created in the standby availability zone.
- The elastic load balancer detects the active gateway replicas in the standby availability zone and redirects the OLTP traffic from the failed primary zone.

In addition to high availability through TiKV replication, TiKV instances are deployed and configured to place each data replica in a different availability zone. As long as two zones are operating normally, the system remains available to serve your application. Data persistence is ensured by regularly pushing data to S3 for high durability. Even if two zones fail, your data on S3 remains accessible and can be recovered.

Applications remain unaffected by non-primary zone failures and are unaware of such issues. In the event of a primary zone failure, both Gateway and TiDB are launched in the standby zone to handle the workload. Applications should implement retry logic for connections to redirect new requests to the active servers in the standby zone.

> 
> 

Reviewer: ```suggestion

```

Applications should handle recoverable failures with retry logic. For specific implementation details and usage, refer to the documentation for your respective driver or ORM (e.g., [JDBC](https://dev.mysql.com/doc/connector-j/en/connector-j-config-failover.html)).

## RTO and RPO

As you develop your business continuity plan, understand the maximum acceptable time before the application fully recovers after the disruptive event. The time required for an application to fully recover is known as the Recovery Time Objective (RTO). Also understand the maximum period of recent data updates (time interval) the application can tolerate losing when recovering from an unplanned disruptive event. The potential data loss is known as Recovery Point Objective (RPO).

The following table compares RPO and RTO of each high availability option:

| High Availability Architecture | RTO (downtime)                | RPO (data loss) |
|--------------------------------|-------------------------------|-----------------|
| Zonal                          | Near 0                        | 0               |
| Regional                       | Typically less than 600 seconds | 0               |

> 

Reviewer: ```suggestion
```

- The Placement Driver (PD) is deployed across multiple availability zones, ensuring high availability by replicating data redundantly across zones.
- Data is replicated across TiKV servers and TiFlash write nodes within the local availability zone.
- TiDB servers and TiFlash compute nodes read from and write to TiKV and TiFlash write nodes, which are safeguarded by storage-level replication.

### Failover process

TiDB Cloud Serverless ensures a transparent failover process for your applications. During a failover:

- A new replica is created to replace the failed one.

- Servers providing storage services recover local caches from persisted data on Amazon S3, restoring the system to a consistent state with the replicas.

In the storage layer, persisted data is regularly pushed to Amazon S3 for high durability. Moreover, immediate updates are not only replicated across multiple TiKV servers but also stored on the EBS of each server, which further replicates the data for additional durability. TiDB automatically resolves issues by backing off and retrying in milliseconds, ensuring the failover process remains seamless for client applications.

> The gateway and computing layers are stateless, so failover involves restarting them elsewhere immediately. Applications should implement retry logic for their connections. While the zonal setup provides high availability, it cannot handle an entire zone failure. If the zone becomes unavailable, downtime will occur until the zone and its dependent services are restored.

Reviewer: ```suggestion

The gateway and computing layers are stateless, so failover involves restarting them elsewhere immediately. Applications should implement retry logic for their connections. While the zonal setup provides high availability, it cannot handle an entire zone failure. If the zone becomes unavailable, downtime will occur until the zone and its dependent services are restored.
```

- **Regional high availability (beta)**: This option distributes nodes across multiple availability zones, offering maximum infrastructure isolation and redundancy. It provides the highest level of availability but requires application-level redundancy across zones. It is recommended to choose this option if you need maximum availability protection against infrastructure failures within a zone. Note that it increases latency and might incur cross-zone data transfer fees. This feature is available in selected regions with multi-availability zone support and can only be enabled during cluster creation. For more information, see [Regional high availability architecture](#regional-high-availability-architecture).

> 
> > **Note:**
> >
> > Regional high availability is in beta and only available in the AWS Tokyo (`ap-northeast-1`) region.

Reviewer: ```suggestion
```
This note is duplicated with https://github.com/pingcap/docs/blob/6cbeec597da53b37635f57acef69a4887841d5d1/tidb-cloud/serverless-high-availability.md?plain=1#L52-L59

The gateway and computing layers are stateless, so failover involves restarting them elsewhere immediately. Applications should implement retry logic for their connections. While the zonal setup provides high availability, it cannot handle an entire zone failure. If the zone becomes unavailable, downtime will occur until the zone and its dependent services are restored.

## Regional high availability architecture

When you create a cluster with regional high availability, critical OLTP (Online Transactional Processing) workload components, such as PD and TiKV, are deployed across multiple availability zones to ensure redundant replication and maximizing availability. During normal operations, components like Gateway, TiDB, and TiFlash compute/write nodes are hosted in the primary availability zone. These components in data plane offer infrastructure redundancy through virtual machine pools, which minimizes failover time and network latency due to colocation.

> **Note:**
>
> - Regional high availability is currently in beta and only available in the AWS Tokyo (`ap-northeast-1`) region.

> > - You can enable regional high availability only during cluster creation.

Reviewer 2: Are there any UI changes for the regional HA? If yes, we also need to update this doc https://docs.pingcap.com/tidbcloud/create-tidb-cluster-serverless

Applications are unaffected by failures in non-primary zones and remain unaware of such events. During a primary zone failure, Gateway and TiDB are launched in the standby availability zone to handle workloads. Ensure that your applications implement retry logic to redirect new requests to active servers in the standby availability zone.

## Automatic backups and durability

Database backups are essential for business continuity and disaster recovery, helping to protect your data from corruption or accidental deletion. With backups, you can restore your database to a specific point in time within the retention period, minimizing data loss and downtime.

TiDB Cloud Serverless provides robust automated backup mechanisms to ensure continuous data protection:

- **Daily full backups**: A full backup of your database is created once a day, capturing the entire database state.
- **Continuous transaction log backups**: Transaction logs are backed up continuously, approximately every 5 minutes, though the exact frequency depends on database activity.

> These automated backups enable you to restore your database either from a full backup or from a specific point in time by combining full backups with continuous transaction logs. This flexibility ensures you can recover your database to a precise point just before an incident occurred.

Reviewer 2: ```suggestion
These automated backups enable you to restore your database either from a full backup or from a specific point in time by combining full backups with continuous transaction logs. This flexibility ensures that you can recover your database to a precise point just before an incident occurs.
```
